plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -1059,9 +1059,7 @@ private function maybe_sync_order( \WC_Abstract_Order &$order, \WC_Abstract_Orde
 		 *
 		 * So we write back to the order table when order modified date is more recent than post modified date. Otherwise, we write to the post table.
 		 */
-		if ( $order_modified_date > $post_order_modified_date ) {
-			return;
-		} else {
+		if ( $post_order_modified_date > $order_modified_date ) {
```

Comment:
I neglected the tests, my fault. But the removal of the '=' comparison is on purpose: if both records have the exact same update date it means that they are in sync and thus there is no need to migrate, right?

Reply:
After discussing with @vedanshujain I'm applying this change. Turns out that data comparison is needed too when the dates are equal, since data in the metadata table could have been modified as well.